esx_service/utils/auth_api.py

```diff
@@ -189,8 +211,12 @@ def modify_privileges(privileges, allow_create, volume_maxsize_in_MB, volume_tot
     """ Modify privileges based on input params """
     logging.debug("modify_privileges: allow_create=%s, volume_maxsize_in_MB=%s, volume_totalsize_in_MB=%s",
                   allow_create, volume_maxsize_in_MB, volume_totalsize_in_MB)
+
+    # If None, don't change the privilege
+    # If not None, change accordingly
     if allow_create is not None:
-        if allow_create == "True":
+        # allow_create has been validated. It is either True or False
+        if allow_create:
```

Comment:
the allow_create is bool here, we can have the explicit bool True check here instead of ```if allow_create:```